### PR TITLE
Flow types

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

**Comment by TrySound** [comment]
Added simple types. Others require some work.

**Comment by renatorib** [comment]
Hey, thanks in advance! I'll take a look tonight 😄

**Comment by renatorib** [comment]
@TrySound it seems good! Nice work

I do not have flow proficiency, and I'm afraid that the flow types be a blocking to me creating new features and components. So I removed typecheck from posttest and prepublishOnly scripts.

Btw, what the pipes `|` mean in flow? I've never seen this syntax.
```
type Foo = {| bar: string |}
                ^               ^
```

**Comment by TrySound** [comment]
@renatorib not extensible object to prevent passing unused field.

**Comment by TrySound** [comment]
Btw, how types will block you since they are completly separated?

**Comment by TrySound** [comment]
I mean at least posttest can stay to prevent merging bad prs with types.

File: package.json [hunk 1]
```diff
@@ -8,37 +8,28 @@
...
-  ],
-  "keywords": [
-    "react",
-    "reactjs",
-    "components",
-    "dumb",
-    "state"
-  ],
+  "files": ["dist", "lib", "src", "es"],
+  "keywords": ["react", "reactjs", "components", "dumb", "state"],
```
Comment (TrySound) [inline package.json]: Do you have automatic formatter for this?

**Comment by renatorib** [comment]
> Btw, how types will block you since they are completly separated?

In case I broke some type changing de API, tests will fail. Publish will fail. And I do not feel able to solve problems of flow.

File: package.json (hunk 1 above)
Comment (renatorib) [reply package.json]: I use prettier in my editor 😅

Comment (renatorib) [reply package.json]: I need to create a .prettierrc file

**Comment by renatorib** [comment]
I have plan to add TS types. When that happens, I'll give more attention to flow/ts testing. 
Also I still need to learn about flow

File: package.json (hunk 1 above)
Comment (TrySound) [reply package.json]: +1

**Comment by TrySound** [comment]
Are you gonna merge this?

**Comment by TrySound** [comment]
Done.
`Compose` component is not typable, so I just added stub with `any`.

**Comment by renatorib** [comment]
Yes, I'll merge this pr soon! Sorry for the delay.
Also, thanks for this, nice work!

**Comment by renatorib** [comment]
Merged! I'll try to publish new version soon

**Pull request merged**